jujugui/static/gui/src/app/components/header-search/test-header-search.js

```diff
@@ -88,6 +88,34 @@ describe('HeaderSearch', function() {
     });
   });
 
+  it('gets cleared when closed', function() {
+    var getAppState = sinon.stub();
+    getAppState.withArgs(
+      'current', 'sectionC', 'component').returns('charmbrowser');
+    getAppState.withArgs('current', 'sectionC', 'metadata')
+               .onFirstCall().returns({ search: 'hexo' });
+    getAppState.withArgs('current', 'sectionC', 'metadata')
+               .onSecondCall().returns({ search: '' });
+    var changeState = sinon.stub();
+    var renderer = jsTestUtils.shallowRender(
+      <juju.components.HeaderSearch
+        getAppState={getAppState}
+        changeState={changeState} />, true);
+    var output = renderer.getRenderOutput();
+    // The input should have the metadata search value
+    var input = output.props.children[0].props.children[1];
+    assert.equal(input.props.value, 'hexo');
+    // re-render which will get the new state.
+    renderer = jsTestUtils.shallowRender(
```

Comment:
Does this need to be `renderer.render(` to update the values rather than render as new?